hardhat.config.ts

```diff
@@ -0,0 +1,168 @@
+// Native
+import { readFileSync } from "fs";
+
+// Vendor
+import Joi from "joi";
+import toml from "toml";
+import "@nomiclabs/hardhat-waffle";
+import "@nomiclabs/hardhat-ethers";
+import "@nomiclabs/hardhat-etherscan";
+import "hardhat-gas-reporter";
+import "solidity-coverage";
+import "hardhat-contract-sizer";
+import { HardhatUserConfig, subtask } from "hardhat/config";
+import { TASK_COMPILE_SOLIDITY_GET_SOURCE_PATHS } from "hardhat/builtin-tasks/task-names";
+
+// Utilities
+import { validateEnvConfig } from "./scripts/utilities/validateConfig";
+
+// Validate values from .env file
+const {
+  FORKING,
+  FORK_BLOCK,
+  SOLC_DEFAULT,
+  PRIVATE_KEY,
+  MNEMONIC,
+  ETH_RPC_URL,
+  ARBITRUM_RPC_URL,
+  CMC_API_KEY,
+  ETHERSCAN_API_KEY,
+} = validateEnvConfig(
+  Joi.object({
```

Comment:
A bit eloborate but a nice way of making sure the required environment variables have been set